crypto/rand/drbg_lib.c

```diff
@@ -501,31 +647,43 @@ void *RAND_DRBG_get_ex_data(const RAND_DRBG *drbg, int idx)
  */
 
 /*
- * Initializes the DRBG with default settings.
- * For global DRBGs a global lock is created with the given name
+ * Initializes the given global DRBG with default settings.
+ * A global lock for the DRBG is created with the given name.
+ *
  * Returns 1 on success, 0 on failure
  */
-static int drbg_setup(RAND_DRBG *drbg, const char *name)
+static int drbg_setup(RAND_DRBG *drbg, const char *name, RAND_DRBG *parent)
 {
     int ret = 1;
 
-    if (name != NULL) {
-        if (drbg->lock != NULL) {
-            RANDerr(RAND_F_DRBG_SETUP, ERR_R_INTERNAL_ERROR);
-            return 0;
-        }
+    if (name == NULL || drbg->lock != NULL) {
+        RANDerr(RAND_F_DRBG_SETUP, ERR_R_INTERNAL_ERROR);
+        return 0;
+    }
 
-        drbg->lock = CRYPTO_THREAD_glock_new(name);
-        if (drbg->lock == NULL) {
-            RANDerr(RAND_F_DRBG_SETUP, RAND_R_FAILED_TO_CREATE_LOCK);
-            return 0;
-        }
+    drbg->lock = CRYPTO_THREAD_glock_new(name);
+    if (drbg->lock == NULL) {
+        RANDerr(RAND_F_DRBG_SETUP, RAND_R_FAILED_TO_CREATE_LOCK);
+        return 0;
     }
 
     ret &= RAND_DRBG_set(drbg,
                          RAND_DRBG_NID, RAND_DRBG_FLAG_CTR_USE_DF) == 1;
     ret &= RAND_DRBG_set_callbacks(drbg, rand_drbg_get_entropy,
                                    rand_drbg_cleanup_entropy, NULL, NULL) == 1;
+
+    if (parent == NULL) {
+        drbg->reseed_interval = MASTER_RESEED_INTERVAL;
+        drbg->reseed_time_interval = MASTER_RESEED_TIME_INTERVAL;
+    } else {
+        drbg->parent = parent;
+        drbg->reseed_interval = SLAVE_RESEED_INTERVAL;
+        drbg->reseed_time_interval = parent->reseed_time_interval;
```

Comment:
This is to ensure that the we actually call the parent DRBG when his time is up? This is non-obvious and could use some comment.

Reply:
Yes this precisely the point. It fixes the first issue mentioned in the commit message of b1979bf77d718d16e04b30b00ef96c9193d9c2f4. But since that commit and its message will be squashed, it's better to have some explaining comments here and in rand_lcl.h, as you suggested. I will add them soon.